### PR TITLE
test: build out integration-test harness (spec phases 1.3, 2, 4)

### PR DESCRIPTION
**Pull request by daniel-rck**
## Summary

Closes the integration-test gap flagged in `docs/specs/integration-tests.md` (audit fix #18 / architecture-review H3). Adds 34 new tests across the WebSocket layer, plugin runtime, plugin spec runner, and download lifecycle — all green on `cargo test --workspace`.

- 5 WebSocket tests on `/api/v1/ws` covering upgrade, `added` / `status_changed` broadcasts, multiple subscribers, and clean disconnect.
- 24 plugin-runtime tests across discovery + transpilation (12), sandbox enforcement (6), and host-API behaviour against wiremock (6).
- 2 spec-runner tests executing the shipped `plugin.spec.ts` files for `generic-http` and `youtube`. Network-dependent YouTube checks are only strict under `AMIGO_NETWORK_TESTS=1`.
- 3 download-lifecycle tests driving a real `Coordinator` against a wiremock backend (end-to-end download, permanent failure, cancel). The retry-after-5xx test is committed `#[ignore]`'d — it tripped audit #13 (per-attempt cancel oneshot is recreated and immediately dropped, self-cancelling retries).
- Replaces the empty `tests/integration/.gitkeep` / `tests/plugins/.gitkeep` placeholders with `tests/README.md` pointing at the established per-crate layout, and marks `docs/specs/integration-tests.md` as `Implemented` with concrete counts.

## Per-phase mapping

| Spec phase | File | New tests |
|---|---|---|
| 1.3 — WebSocket | `crates/server/tests/ws_tests.rs` | 5 |
| 2.1 — Loader / transpiler | `crates/plugin-runtime/tests/runtime_tests.rs` | 12 |
| 2.2 — Sandbox enforcement | `crates/plugin-runtime/tests/sandbox_tests.rs` | 6 |
| 2.3 — Host API via wiremock | `crates/plugin-runtime/tests/host_api_tests.rs` | 6 |
| 2.4 — Spec-file runner | `crates/plugin-runtime/tests/spec_runner.rs` | 2 |
| 4 — Download lifecycle | `crates/core/tests/download_lifecycle_tests.rs` | 3 (+1 ignored) |

Phases 1.1 / 1.2 (server API) and Phase 3 (storage edge cases) were already in `crates/server/tests/api_tests.rs`, `auth_flow_tests.rs`, and `crates/core/tests/storage_tests.rs` before this branch and are left untouched.

## Test plan

- [x] `cargo fmt --all -- --check` — clean
- [x] `cargo clippy --workspace --all-targets -- -D warnings` — clean
- [x] `cargo test --workspace` — all originally-passing tests still green, 34 new tests pass, 1 ignored
- [ ] CI rerun on PR
- [ ] Optional: `AMIGO_NETWORK_TESTS=1 cargo test -p amigo-plugin-runtime --test spec_runner` to verify the strict YouTube path on a runner with internet
- [ ] Once audit #13 is fixed, drop the `#[ignore]` on `test_download_retries_on_transient_5xx`

https://claude.ai/code/session_014r1dSzz2spDTtqeMyyEvwZ

---
_Generated by [Claude Code](https://claude.ai/code/session_014r1dSzz2spDTtqeMyyEvwZ)_

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+directories were already established by the time the spec was written,
+so we kept them.
+
+Where to find tests by area:
+
+| Area | Location | Notable files |
+|---|---|---|
+| REST API + auth + WebSocket | `crates/server/tests/` | `api_tests.rs`, `auth_flow_tests.rs`, `ws_tests.rs` |
+| Coordinator + storage + download lifecycle | `crates/core/tests/` | `coordinator.rs`, `storage_tests.rs`, `download_lifecycle_tests.rs` |
+| Plugin loader, sandbox, host-API, spec runner | `crates/plugin-runtime/tests/` | `runtime_tests.rs`, `sandbox_tests.rs`, `host_api_tests.rs`, `spec_runner.rs` |
```
[the remark this answers is not in the document]
Comment (daniel-rck) [reply tests/README.md:15]: False positive — the table uses single `|` row prefixes (`| Area | Location | Notable files |` followed by `|---|---|---|`), not `||`. That is standard GitHub-flavored Markdown and renders correctly on github.com.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: docs/specs/integration-tests.md [hunk 2]
```diff
@@ -1,13 +1,48 @@
...
+
+| Phase | File | Tests |
+|---|---|---|
+| 1.3 — WebSocket | `crates/server/tests/ws_tests.rs` | 5 |
+| 2.1 — Plugin loader / transpiler | `crates/plugin-runtime/tests/runtime_tests.rs` | 12 |
+| 2.2 — Sandbox enforcement | `crates/plugin-runtime/tests/sandbox_tests.rs` | 6 |
+| 2.3 — Host API via wiremock | `crates/plugin-runtime/tests/host_api_tests.rs` | 6 |
+| 2.4 — `plugin.spec.ts` runner | `crates/plugin-runtime/tests/spec_runner.rs` | 2 |
+| 4   — Download lifecycle | `crates/core/tests/download_lifecycle_tests.rs` | 3 (+1 ignored, see below) |
+| **Total new** |  | **34** |
```
[the remark this answers is not in the document]
Comment (daniel-rck) [reply docs/specs/integration-tests.md:29]: False positive — the table uses single `|` row prefixes (`| Phase | File | Tests |` followed by `|---|---|---|`), not `||`. Standard GitHub-flavored Markdown, renders correctly.

---
_Generated by [Claude Code](https://claude.ai/code)_
